### PR TITLE
Favourite Posts

### PR DESCRIPTION
**Pull request by SubhradeepSS**
Issue fixed: #17 

Features added:

- Add/Remove a post to favourites section

- Separate html template for viewing blogs added to favourites
favourites_in_home
add
delete
favorites_page

### CODE REVIEW

**Comment by SubhradeepSS** [comment]
@diyajaiswal11 can u please review this PR for the **Favourites** feature?

**Comment by SubhradeepSS** [comment]
@diyajaiswal11 I have resolved all merge conflicts. Can u please review my work and suggest any changes?

File: bloggitt/core/views.py [hunk 1]
```diff
@@ -48,9 +48,47 @@ def postlist(request):
...
+    post = Post.objects.get(slug=slug)
+
+    if post not in Favourites.posts.all():
+        Favourites.posts.add(post)
+        Favourites.save()
+    else:
+        Favourites.posts.remove(post)
+        Favourites.save()
+    
+    return redirect('favourites')
```
Comment (diyajaiswal11) [inline bloggitt/core/views.py:83]: Whenever a user adds or removes a post from his favorites, he should stay on the same page and the button should change accordingly. Why are you redirecting him to favourites page? Please look into this.

**Review by diyajaiswal11 — CHANGES_REQUESTED** [review]

File: bloggitt/core/views.py (hunk 1 above)
Comment (SubhradeepSS) [reply bloggitt/core/views.py:83]: Ok, i will update that

Comment (SubhradeepSS) [reply bloggitt/core/views.py]: I have changed the redirect to `post_detail` so as to stay in same page.

File: bloggitt/core/views.py [hunk 2]
```diff
@@ -48,9 +48,47 @@ def postlist(request):
...
-    return render(request, 'detail.html', {'post': post})
+
+    if post not in Favourites.posts.all():
+        Favourites.posts.add(post)
+    else:
+        Favourites.posts.remove(post)
+    
+    Favourites.save()
+    
+    return redirect('post_detail', slug=slug)
```
Comment (diyajaiswal11) [inline bloggitt/core/views.py:83]: According to me, whenever a user clicks on Add to Favorite button, the post is added to his fav and page reloads and button changes to remove from fav, isn't it?? I don't want page reload everytime, the button should change without page reload and adding or removing from fav should also happen.

**Review by diyajaiswal11 — CHANGES_REQUESTED** [review]

File: bloggitt/core/views.py (hunk 2 above)
Comment (SubhradeepSS) [reply bloggitt/core/views.py]: I have used ajax so as not to reload the page, but also doing the backend work successfully.

Comment (diyajaiswal11) [reply bloggitt/core/views.py]: Okayy @SubhradeepSS , Good work!!

**Pull request merged**